### PR TITLE
Implementation of DataArts cluster API.

### PR DESCRIPTION
**Pull request by sgmv**
### What this PR does / why we need it

First changes for DataArt API. This PR brings ability to manage DataArts clusters.

### CODE REVIEW

File: acceptance/openstack/dataarts/v1.1/clusters_test.go [hunk 1]
```diff
@@ -18,19 +18,19 @@ func TestDataArtsClusterLifecycle(t *testing.T) {
...
 
 	dataStore := cluster.Datastore{
 		Type:    "cdm",
-		Version: "2.9.1.200",
+		Version: "2.10.0.100",
 	}
 	instance := cluster.Instance{
 		AZ:        "eu-de-01",
-		FlavorRef: "a79fd5ae-1833-448a-88e8-3ea2b913e1f6",
+		FlavorRef: "5ddb1071-c5d7-40e0-a874-8a032e81a697",
```
Comment (anton-sidelnikov) [inline acceptance/openstack/dataarts/v1.1/clusters_test.go:33]: there is no api for obtain this flavor id?

File: openstack/dataarts/v1.1/cluster/Get.go [hunk 2]
```diff
@@ -24,131 +19,270 @@ func extraResp(err error, raw *http.Response) (*ClusterQuery, error) {
...
+	// Cluster ID
+	Id string `json:"id"`
+	// Whether the cluster is frozen. The value can be 0 (not frozen) or 1 (frozen).
+	IsFrozen string `json:"isFrozen"`
+	// Cluster configuration status. Options: - In-Sync: The cluster configuration has been synchronized. - Applying: The cluster is being configured. - Sync-Failure: The cluster configuration failed.
+	Actions []string `json:"actions"`
+	// Cluster update time in ISO 8601 format: YYYY-MM-DDThh:mm:ssZ
+	Updated string `json:"updated"`
+	// Cluster status
+	//
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1.1/cluster/Get.go]: please delete empty lines

Comment (anton-sidelnikov) [reply openstack/dataarts/v1.1/cluster/Get.go]: and above

File: openstack/dataarts/v1.1/cluster/Get.go [hunk 3]
```diff
@@ -24,131 +19,270 @@ func extraResp(err error, raw *http.Response) (*ClusterQuery, error) {
...
+	// Whether the node is frozen. The value can be 0 (not frozen) or 1 (frozen).
+	IsFrozen string `json:"isFrozen"`
+	// Component
+	Components string `json:"components"`
+	// Node configuration status. The value is null when the cluster list is queried.
+	//
+	//    In-Sync: The configuration has been synchronized.
+	//
+	//    Applying: The configuration is in progress.
+	//
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1.1/cluster/Get.go]: and here

File: openstack/dataarts/v1.1/cluster/Start.go [hunk 4]
```diff
@@ -2,20 +2,36 @@ package cluster
...
 
-func Start(client *golangsdk.ServiceClient, id string) (*JobId, error) {
-	type Start struct {
-		Start *EmptyObj `json:"start"`
+type StartOpts struct {
+	// Start Starting a cluster. This parameter is an empty object.
+	Start EmptyStruct `json:"start"`
+}
+
+type EmptyStruct struct{}
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1.1/cluster/Start.go]: please move this to common/structs

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: acceptance/openstack/dataarts/v1.1/clusters_test.go (hunk 1 above)
Comment (sgmv) [reply acceptance/openstack/dataarts/v1.1/clusters_test.go:33]: Unfortunately, not. They are hardcoded in the documentation.

File: openstack/dataarts/v1.1/cluster/Start.go (hunk 4 above)
Comment (sgmv) [reply openstack/dataarts/v1.1/cluster/Start.go]: done

File: openstack/dataarts/v1.1/cluster/Get.go (hunk 3 above)
Comment (sgmv) [reply openstack/dataarts/v1.1/cluster/Get.go]: done

File: openstack/dataarts/v1.1/cluster/Get.go (hunk 2 above)
Comment (sgmv) [reply openstack/dataarts/v1.1/cluster/Get.go]: done

**Review by anton-sidelnikov — APPROVED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**